src/parseTools.js

```diff
@@ -197,10 +197,20 @@ function isStructPointerType(type) {
   return !Compiletime.isNumberType(type) && type[0] == '%';
 }
 
+const POINTER_SIZE = MEMORY64 ? 8 : 4;
+const POINTER_BITS = POINTER_SIZE * 8;
+const POINTER_TYPE = 'i' + POINTER_BITS;
+
+const SIZE_TYPE = POINTER_TYPE;
+
 function isPointerType(type) {
   return type[type.length - 1] == '*';
 }
 
+function sizeT(x) {
+  return MEMORY64 ? `BigInt(${x})` : x;
```

Comment:
It looks like `MEMORY64` assumes `WASM_BIGINT`? That seems reasonable, but is it documented somewhere? Also it would be good to assert on that.

Reply:
Yes.
`emcc.py` contains:
```
  # Any "pointers" passed to JS will now be i64's, in both modes.
  if settings.MEMORY64:
    if settings_map.get('WASM_BIGINT') == '0':
      exit_with_error('MEMORY64 is not compatible with WASM_BIGINT=0')
    settings.WASM_BIGINT = 1
```
Is that sufficient? Where should it be documented?

Reply:
Documenting it in settings.js on the setting sgtm.